web/src/pages/billing/SequencingGroupsByMonth.tsx

```diff
@@ -0,0 +1,263 @@
+const SequencingGroupsByMonth: React.FunctionComponent = () => {
+    // Use navigate and update url params
+    const navigate = useNavigate()
+    const { projectName } = useParams()
+
+    // Data loading
+    const [isLoading, setIsLoading] = React.useState<boolean>(true)
+    const [error, setError] = React.useState<string | undefined>()
+    const [message, setMessage] = React.useState<string | undefined>()
+    const [data, setData] = React.useState<TypeData[]>([])
+
+    // Chart refs
+    const containerRef = useRef<HTMLDivElement>(null)
+    const [measureRef, { width, height }] = useMeasure<HTMLDivElement>()
+
+    // Callback to get data for plotting.
+    const onProjectSelect = (_project: IMetamistProject) => {
+        navigate(`/billing/sequencingGroupsByMonth/${_project.name}`)
+    }
+
+    const getData = React.useCallback(async (project: string | undefined) => {
+        if (project == undefined) {
+            return
+        }
+
+        setIsLoading(true)
+        setError(undefined)
+        setMessage(undefined)
+
+        try {
+            const sg_api = new SequencingGroupApi()
+            const result = await sg_api.sequencingGroupHistory(project)
+            setIsLoading(false)
+
+            const resultData = result.data as ProjectHistory
+            const newDataset: TypeData[] = []
+            for (const [date, counts] of Object.entries(resultData)) {
+                for (const [typeName, count] of Object.entries(counts)) {
+                    newDataset.push({
+                        date: new Date(date),
+                        type: typeName,
+                        count: count,
+                    })
+                }
+            }
+
+            setData(newDataset)
+        } catch (er: any) {
+            setIsLoading(false)
+            setError(er.message)
+        }
+    }, [])
+
+    // Memo of debounced getData so that the debounce function will only be regenerated when the project changes.
+    const debouncedGetData = React.useMemo(() => {
+        return debounce(() => {
+            getData(projectName)
+        }, 1000)
+    }, [projectName, getData])
+
+    // Cleanup debounced function on unmount.
+    React.useEffect(() => {
+        return () => {
+            debouncedGetData.cancel()
+        }
+    }, [projectName, debouncedGetData])
+
+    // Effect to trigger data retrieval on initial load and project value changes.
+    React.useEffect(() => {
+        debouncedGetData()
+    }, [debouncedGetData])
+
+    React.useEffect(() => {
+        if (!data) return
+
+        const plot = Plot.plot({
+            color: {
+                scheme: 'spectral',
+                legend: true,
+            },
+            x: { interval: 'month' },
+            y: { grid: true },
+            width,
+            height,
+            marks: [
+                Plot.barY(
+                    data,
+                    Plot.stackY({
```

Comment:
One thing I noticed in testing is that the bars don't always stack in a consistent order. I think you can avoid this by setting a `order` property on the `stackY` config. eg: `order: '-sum'`